ethers-solc/src/resolver/mod.rs

```diff
@@ -627,18 +627,22 @@ impl VersionedSources {
 
         let mut sources_by_version = std::collections::BTreeMap::new();
         for (version, sources) in self.inner {
+            let mut solc = None;
             if !version.is_installed() {
                 if self.offline {
                     return Err(SolcError::msg(format!(
                         "missing solc \"{}\" installation in offline mode",
                         version
                     )))
                 } else {
-                    Solc::blocking_install(version.as_ref())?;
+                    solc = Some(Solc::blocking_install(version.as_ref())?);
```

Comment:
Tiny point:

Why do we use an `Option<T>` here?

We could even do:
```suggestion
                    solc = Solc::blocking_install(version.as_ref());
```

and then:
```rust
let solc = solc.unwrap_or_else(|| {
    Solc::find_svm_installed_version(version.to_string())?.ok_or_else(|| {
        SolcError::msg(format!("solc \"{}\" should have been installed", version))
    })
})
```

Reply:
hmm this way we need another else branch, which is fine I guess, will add